velox/connectors/hive/storage_adapters/s3fs/S3FileSystem.cpp

```diff
@@ -506,12 +508,14 @@ void finalizeS3() {
 
 class S3FileSystem::Impl {
  public:
-  Impl(const Config* config) : config_(config) {
+  Impl(const Config* config) {
+    hiveConfig_ = std::make_shared<HiveConfig>(
+        std::make_shared<core::MemConfig>(config->values()));
```

Comment:
config is nullptr here.

FileSystem was created by FileHandleGenerator, but FileHandleGenerator's properties was set to nullptr in HiveConnector ctor.